enhancements/api-review/custom-tags-aws.md

```diff
@@ -0,0 +1,144 @@
+ - https://github.com/openshift/api/pull/864
+ - https://github.com/openshift/cluster-ingress-operator/pull/578
+
+## Motivation
+
+Motivations include but are not limited to:
+
+ - allow admin, compliance, and security teams to keep track of assets and objects created by OpenShift,
+   both at install time during continuous operation (Day 2)
+ - in a Managed OpenShift environment such as Red Hat OpenShift on AWS (ROSA), allow easy differentiation
+   between Red Hat-managed AWS objects and customer-managed objects
+ - *allow for the restriction of permissions granted to Red Hat in an AWS account by AWS resource tags.
+   For example, see SDE-1146 - "IAM users and roles can only operate on resources with specific tags"*
+
+### Goals
+
+ - the administrator or service (in the case of Managed OpenShift) installing OpenShift can pass an arbitrary
+   list of user-defined tags to the OpenShift Installer, and everything created by the installer and all other
+   bootstrapped components will apply those tags to all resources created in AWS, for the lift of the cluster.
+ - Installer writes the tags to the infrastructure resource so that Day 2 operations can know what tags to apply
+   to AWS resources for the life of the cluster.
+ - tags must be applied at creation time, in an atomic operation. It isn't acceptable to create an object and,
+   after some period of time, apply the tags post-creation.
+
+### Non-Goals
+
+ - to reduce initial scope, tags are applied only at creation time and not reconciled. If an administrator manually
+   changes the tags stores in the infrastructure resource, pre-existing AWS resources are not updated. Only newly-
+   created AWS resources would get the updated tags.
+ - to reduce initial scope, we're not implementing this for clouds other than AWS. We shouldn't take any actions to
+   prohibit that later, though.
+
+## Proposal
+
+Add a new field `userTags` to `.spec.aws` of the `infrastructure.config.openshift.io` type. Tags included in the
+`userTags` field will be applied to new resources created for the cluster.
+
+Installer will apply these tags to all AWS resources it creates with terraform (e.g. bootstrap and master EC2 instances)
+
+All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api)
+will apply these tags to all AWS resources they create.
+
+### User Stories
+
+SDE-1146 - IAM users and roles can only operate on resources with specific tags
+As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
+AWS resource tags.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Some components, like machine-api and image registry, already have some support to allow adminstrators to
+push tags into EC2 instance and S3 buckets, respectively.
+
+In the case of machine-api, a user has to specify tags on the MachineSet as part of the Day 2 operation.
```

Comment:
This is untrue.  Users can modify machinesets during the installation process.

Reply:
It's also important to note here that the installer creates machinesets as part of the install process so _could_ populate the correct fields when it does that

Reply:
In fact, the installer already **does** populate the machines and machinesets that it creates with the additional user-supplied tags.

Reply:
Does that mean there could be conflicts if installer created Machinesets have the same set of tags as going into the infrastructure spec? Or does it populate the tags on the machinesets from a different part of the install config?

Reply:
removed this section